tla-io/src/main/scala/at/forsyte/apalache/io/ConfigManager.scala

```diff
@@ -50,67 +34,74 @@ trait CliConfig {
  * @param cmd
  *   The configurations supplied via CLI
  */
-case class ConfigManager(cmd: CliConfig) {
+class ConfigManager() {
   private val TLA_PLUS_DIR = ".tlaplus"
   private val APALACHE_CFG = "apalache.cfg"
   private val DOT_APALACHE_CFG = "." + APALACHE_CFG
 
   import Converters._
 
   // Recursively search parents of given `dir`, looking for .apalache.cfg file
-  private def findLocalConfig(dir: Path): Option[ConfigObjectSource] = {
+  private def findLocalConfig(dir: Path): Option[File] = {
     val localCfg = dir.resolve(DOT_APALACHE_CFG)
     if (Files.exists(localCfg)) {
-      Some(ConfigSource.file(localCfg))
+      Some(localCfg.toFile())
     } else {
       Option(dir.getParent).flatMap(findLocalConfig)
     }
   }
 
-  private def localConfig(): Option[ConfigObjectSource] = {
-    cmd.configFile.map(ConfigSource.file).orElse(findLocalConfig(Paths.get(".").toAbsolutePath()))
-  }
-
   /**
-   * Load the application configuration from all sources supported by apalache
+   * Load the application configuration from all sources supported by apalache, and merge with the primary `cfg`
    *
    * The following precedence is maintained, wherein configured values found from lower numbered sources override values
    * from higher numbered sources:
    *
-   *   1. CLI arguments
-   *   1. Environment variables (Overiding is taken care of by the CLI parsing library)
+   *   1. The primary `cfg` supplied as a parameter (typically derived from the CLI)
    *   1. Local config file
    *   1. Global config file
    *   1. `ApalacheConfig` defaults (as specified in the case class definition)
+   *
+   * @param cfg
+   *   The primary configuration, which will be used to override values from other configs.
    */
-  def load(): ConfigReader.Result[ApalacheConfig] = {
+  def load(cfg: ApalacheConfig): ConfigReader.Result[ApalacheConfig] = {
 
     val home = System.getProperty("user.home")
     val globalConfig = ConfigSource.file(Paths.get(home, TLA_PLUS_DIR, APALACHE_CFG))
 
-    localConfig()
-      .getOrElse(ConfigSource.empty)
-      // `withFallback` supplies configuration sources that only apply if the preceding configs aren't set
-      .withFallback(globalConfig.optional)
+    val localConfig: ConfigObjectSource =
+      cfg.common.configFile.orElse(findLocalConfig(Paths.get(".").toAbsolutePath())) match {
+        case Some(cfgFile) => ConfigSource.file(cfgFile)
+        case None          => ConfigSource.empty
+      }
+
+    // Derive lightbend `Config` objects from the default and primary config
+    // case classes.
```

Comment:
What are the default and primary config case classes?

Reply:
They are the named variable commented here: 

- The "primary config" is derived from the `cfg` param supplied to this function (see the `@param cfg` in the scaladoc above).
- The default is derived from the default values specified in `ApalacheConfig`, invoked by calling applying the case class.

Reply:
Oh, I missed that this refers to the doccomment on `load`. All good!